apps/workshop/homarr_provider_test.go

```
@@ -0,0 +1,270 @@
+	payload["messages"].([]any)[0].(map[string]any)["content"] = append(
+		payload["messages"].([]any)[0].(map[string]any)["content"].([]any),
+		map[string]any{"type": "image_url", "image_url": map[string]any{"url": "data:image/png;base64,YQ=="}},
+	)
+	if err := validateProviderInput(payload); !errors.Is(err, errTooManyImages) {
+		t.Fatalf("expected six uploaded images to be rejected, got %v", err)
+	}
+}
+
+func TestProviderInFlightLimits(t *testing.T) {
+	provider := &homarrProvider{inFlightByUser: make(map[string]int)}
+	if !provider.acquireRequest("one") || !provider.acquireRequest("one") || provider.acquireRequest("one") {
```

golangci-lint (2.12.2)

[error] SA4000: identical expressions on the left and right side of the '||' operator

(staticcheck)